### PR TITLE
Web3auth starter kit (WIP)

### PR DESCRIPTION
**Pull request by webthethird**
## Description

Integrates Web3Auth login UX. This is currently still a work in progress, and I would like some help tightening up the integration. 

Should not be merged into main, but rather a `scaffold-eth-2/web3auth-starter-kit` branch, which I would create if I could.

## Additional Information

There are multiple ways to integrate Web3Auth, and I have tried the two below with the following reasons:

1. Integrate into the Rainbow Kit connect modal using the existing custom connect button
    - This seems like a cleaner UX option, and it integrates nicely with the rest of Scaffold-ETH
    - Based on the following [example](https://github.com/Web3Auth/web3auth-pnp-examples/tree/main/wagmi-connector/rainbowkit-react-modal-example)
    - Only requires a new [`Web3AuthConnectorInstance`](https://github.com/webthethird/scaffold-eth-2/blob/web3auth-starter-kit/packages/nextjs/components/scaffold-eth/Web3AuthConnectorInstance.tsx) component and using it in [`WagmiConnectors`](https://github.com/webthethird/scaffold-eth-2/blob/056d1f2f9a27df981845deae9b8fd11f0db36ff9/packages/nextjs/services/web3/wagmiConnectors.tsx#L25)
2. Use Web3Auth directly to gain access to the full set of features
    - Based on the following [example](https://github.com/Web3Auth/web3auth-pnp-examples/tree/main/web-modal-sdk/evm/react-evm-modal-example)
    - According to [this post](https://web3auth.io/community/t/extract-web3auth-user-information-while-using-wagmi-and-rainbowkit/4837/4?u=w3bthe3rd) in their community support forum, using the wagmi connector makes it impossible to use any of the Web3Auth features such as `getUserInfo()` and `authenticateUser()`.
    - For instance, if I'm a mainstream user, I want to see my email address displayed in the ConnectButton once I'm logged in, not an Ethereum address. And I'd rather see my profile image instead of a blockie.
    - ~~However, I am having some difficulty figuring out how to integrate the account provided by Web3Auth with the rest of Scaffold-ETH, i.e., how to get the faucet button to send funds to the correct address, since this solution doesn't use wagmi. I am still relatively new to Web3 front-end development (I'm more of a smart contract dev still).~~

~~I would love to get some help / advice on finding a solution that gives us the best of both approaches, i.e., tighter integration with the rest of Scaffold-ETH 2 while providing access to the full set of features provided by using Web3Auth directly.~~

Edit: Got it working with the benefits of both approaches, with some tweaks to WagmiConnectors.tsx and RainbowKitCustomConnectButton.tsx :tada:

- [X] I have read the [contributing docs](/scaffold-eth/scaffold-eth-2/blob/main/CONTRIBUTING.md) (if this is your first contribution)
- [X] This is not a duplicate of any [existing pull request](https://github.com/scaffold-eth/scaffold-eth-2/pulls)

## Related Issues

#449 

Your ENS/address:

WEBthe3rd.eth

### CODE REVIEW

File: packages/nextjs/services/web3/wagmi-burner/BurnerConnectorErrors.ts
Comment (webthethird) [inline packages/nextjs/services/web3/wagmi-burner/BurnerConnectorErrors.ts:1]: This was due to a linting issue, automatically changed by `yarn format`

**Comment by webthethird** [comment]
One thing I could still use some help with is finding a way to keep the Web3Auth login modal from popping up as soon as you enter the site.

**Comment by webthethird** [comment]
Also, let me know which branch to switch the base to! Or feel free to change it for me :slightly_smiling_face:

**Comment by carletex** [comment]
Hey @webthethird 

I watched your presentation on BTF and it was pretty cool. Thanks for this!

In SE-1, there are hundreds of branches that people contributed, showcasing a huge variety of use cases for web3 apps that people could use as templates.

In SE-2, we didn't embrace the same model, since we don't think branches are meant for that. They are now primarily used for isolating feature development & bug fixing. This helps us maintain a clear and organized development workflow.

We don't have a great system for showcasing community builds right now (but it's on the roadmap!**). I think the best you can do is create your build in your BG profile (https://app.buidlguidl.com/) so people can find it in the /builds route. 

** The idea is to improve the current /builds page, with tagging / filter search / etc.

Thanks again!

**Comment by webthethird** [comment]
Hey @carletex! That makes sense :+1: I used SE-1 in the past, including various starter kit branches, so that's where I got the idea that this would live on a branch in the main repo. But as someone who recently launched an open-source project of my own, I totally get the decision to limit branch usage to bug fixes and core feature development.

I'll close the PR here, and I've already added it to my BG profile [here](https://app.buidlguidl.com/builders/0x830c91140AD5851D5b765691e14c04101829Eaea). I wonder though, do you have any feedback on the implementation (or an idea for how to keep that Web3Auth modal from popping up automatically on page load)? If so, I wonder if it's still possible to leave a code review for me, either on this (now closed) PR or over on my fork. Any and all feedback would be very welcome!

**Pull request closed without merging**